SocialCareCaseViewerApi.Tests/V1/UseCase/WarningNoteUseCaseTests.cs

```diff
@@ -0,0 +1,136 @@
+                    It.IsAny<CreateWarningNoteRequest>()))
+                .Returns(responseObject);
+
+            var response = _classUnderTest.ExecutePost(new CreateWarningNoteRequest());
+
+            response.Should().NotBeNull();
+            response.Should().BeEquivalentTo(responseObject);
+        }
+
+        [Test]
+        public void ExecutePostCallsDatabaseGatewayWithParametersOneTime()
+        {
+            var request = _fixture.Create<CreateWarningNoteRequest>();
+
+            _classUnderTest.ExecutePost(request);
+
+            _mockDatabaseGateway.Verify(x => x.CreateWarningNote(request),
+                                    Times.Once);
+        }
+
+        [Test]
+        public void ExecutePostReturnsCorrectCaseNoteId()
+        {
+            var request = new CreateWarningNoteRequest();
+
+            var expectedResponse = new CreateWarningNoteResponse() { CaseNoteId = _fixture.Create<string>() };
+
+            _mockDatabaseGateway
+                .Setup(x => x.CreateWarningNote(
+                    It.Is<CreateWarningNoteRequest>(x => x == request)))
+                .Returns(expectedResponse);
+
+            var response = _classUnderTest.ExecutePost(request);
+
+            response.CaseNoteId.Should().BeEquivalentTo(expectedResponse.CaseNoteId);
+        }
+        #endregion
+
+        #region Get Warning Note
+        [Test]
+        public void ExecuteGetReturnsACreateWarningNoteResponse()
+        {
+            _mockDatabaseGateway.Setup(
+                x => x.GetWarningNotes(It.IsAny<GetWarningNoteRequest>()))
+                .Returns(new List<dbWarningNote>());
+
+            var response = _classUnderTest.ExecuteGet(new GetWarningNoteRequest());
+
+            response.Should().NotBeNull();
+            response.Should().BeOfType<GetWarningNoteResponse>();
+        }
+
+        [Test]
+        public void ExecuteGetCallsTheDatabaseGateWayWithAParameter()
+        {
+            _mockDatabaseGateway.Setup(
+                x => x.GetWarningNotes(It.IsAny<GetWarningNoteRequest>()))
+                .Returns(new List<dbWarningNote>());
+
+            var request = new GetWarningNoteRequest();
+
+            _classUnderTest.ExecuteGet(request);
+            _mockDatabaseGateway.Verify(
+                x => x.GetWarningNotes(request), Times.Once);
+        }
+
+        [Test]
+        public void ExecuteGetReturnsTheExpectedResponse()
+        {
+            var stubbedWarningNote = _fixture.Build<dbWarningNote>()
+                                    .Without(x => x.Person)
+                                    .Create();
+            var stubbedList = new List<dbWarningNote>
+            {
+                stubbedWarningNote
+            };
+
+            _mockDatabaseGateway.Setup(
+                x => x.GetWarningNotes(It.IsAny<GetWarningNoteRequest>()))
+                .Returns(stubbedList);
+
+            var expectedResponse = new GetWarningNoteResponse
```

Comment:
You don't need to change this but it would be tidier if the response was just `List<WarningNote>` instead of `GetWarningNoteResponse`.

`List<WarningNote>` actual response to the frontend would look like [warningNote1, warningNote2, warningNote3...]

`GetWarningNoteResponse` is an object that contains a single field `WarningNotes` which then has the list of warning notes so something like {WarningNotes: [warningNote1, warningNote2, warningNote3...]}.

It's basically a useless object wrapping a single field which is the list we care about.

Reply:
Removed GetWarningNoteResponse following discussion